homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -106,8 +161,10 @@
         vol.Optional(CONF_DEVICE_CLASS): cv.string,
         vol.Required(CONF_OBSERVATIONS): vol.Schema(
             vol.All(
+                [vol.Any(TEMPLATE_SCHEMA, STATE_SCHEMA, NUMERIC_STATE_SCHEMA)],
+                _above_greater_than_below,
```

Comment:
So this seems to be working, however the tests are failing because the exceptions don't seem to be making it into the logs. I think this is because the failure of NUMERIC_STATE_SCHEMA causes an error of PLATFORM_SCHEMA.

A simple workaround would be to call _LOGGER.error from _abive_greater_than_below directly, but I wanted to check if this is a symptom of something I've done wrong

Reply:
Okay, fixed in https://github.com/home-assistant/core/pull/119281/commits/05b4f6b15fdc904a65cf3feebd7c300ab446b48e, because each schema is checked within a `vol.any()` block raising the `vol.Invalid("")` doesn't show up in the logs,. So instead I now directly send the error to the logs and raise a simpler `vol.Invalid`